### PR TITLE
SubmitEvent.submitter & FormData(form, submitter)

### PR DESCRIPTION
**Pull request by filiphlm**
Solves #401. No polyfill(s) included.

### CODE REVIEW

File: src/core/UIHandler.ts [hunk 1]
```diff
@@ -20,41 +20,30 @@ export class UIHandler extends EventTarget {
...
-		const elements = element.querySelectorAll(selectors);
-		elements.forEach((element) => bindElement(element));
+		const elements = element.querySelectorAll(selector);
+		elements.forEach((element) => bindElement(element as HTMLAnchorElement));
 
-		if (element.matches(selectors)) {
-			bindElement(element);
+		if (element.matches(selector)) {
+			bindElement(element as HTMLAnchorElement);
 		}
```
Comment (filiphlm) [inline src/core/UIHandler.ts:35]: BTW: This looks suspicious to me. Shouldn't it be more like this?

```typescript
if (element.matches(selectors)) {
	bindElement(element);
} else {
	const elements = element.querySelectorAll(selectors);
	elements.forEach((element) => bindElement(element));
}
```
(The same applies to form binding below.)

**Review by jiripudil — CHANGES_REQUESTED** [review]
Thanks, I really like how this simplifies things. I just have a few suggestions regarding the BC and ergonomics of the API

File: src/core/UIHandler.ts (hunk 1 above)
Comment (filiphlm) [reply src/core/UIHandler.ts:32]: Done.

File: src/core/UIHandler.ts [hunk 2]
```diff
@@ -73,52 +62,27 @@ export class UIHandler extends EventTarget {
...
-		return this.processInteraction(element, method, url, data, options, event);
+	public async clickElement(element: HTMLAnchorElement, options: Options = {}, event?: MouseEvent): Promise<Payload> {
+		return this.processInteraction(element, 'GET', element.href, null, options, event);
 	}
 
-	public async submitForm(form: HTMLFormElement, options: Options = {}, event?: Event): Promise<Payload> {
-		const method = form.getAttribute('method')?.toUpperCase() ?? 'GET';
-		const url = form.getAttribute('action') ?? window.location.pathname + window.location.search;
-		const data = new FormData(form);
+	public async submitForm(form: HTMLFormElement, options: Options = {}, event?: SubmitEvent): Promise<Payload> {
```
[the remark this answers is not in the document]
Comment (filiphlm) [reply src/core/UIHandler.ts]: I tried: https://github.com/naja-js/naja/pull/402/commits/1b0b2d2518a81aadf49913515c8d218441411d1b

The bottom line is: No `SubmitEvent`, no `SubmitEvent.submitter`. If I'm not mistaken, with a missing or non-standard event (ie anything other than `SubmitEvent`) there will be no change from the previous (pre-PR) behavior. Are you OK with that?

File: src/core/UIHandler.ts [hunk 3]
```diff
@@ -73,52 +62,27 @@ export class UIHandler extends EventTarget {
...
-			} else if (element.type === 'image') {
-				const coords = element.getBoundingClientRect();
-				const prefix = element.name !== '' ? `${element.name}.` : '';
-				data.append(`${prefix}x`, Math.max(0, Math.floor(event !== undefined ? event.pageX - coords.left : 0)));
-				data.append(`${prefix}y`, Math.max(0, Math.floor(event !== undefined ? event.pageY - coords.top : 0)));
-			}
-		}
-
-		return this.processInteraction(element, method, url, data, options, event);
+	public async clickElement(element: HTMLAnchorElement, options: Options = {}, event?: MouseEvent): Promise<Payload> {
```
[the remark this answers is not in the document]
Comment (filiphlm) [reply src/core/UIHandler.ts]: [Maybe this way?](https://github.com/naja-js/naja/pull/402/commits/dfe8ddebc26935aeb5d65616fc296858d75c0f52)

**Review by jiripudil — APPROVED** [review]

**Comment by jiripudil** [comment]
Hi, sorry this took me so long. I've been having a downswing in terms of time and energy that I'm willing to dedicate to OSS recently.

While reviewing the code, I've had some more thoughts on how to improve its type safety. I took the liberty of implementing the ideas myself, because I didn't want you to have to wait another month for me to come back to this.

I really appreciate the good work you've put into this, and I'm looking forward to your future contributions!

**Pull request merged**